adding_species/contributing.ipynb

```diff
@@ -0,0 +1,990 @@
+    "information, genetic maps and simulation models from the literature.\n",
+    "NOTE: By default, the tskit '.trees' binary file is written to\n",
+    "stdout,so you should either redirect this to a file or use the '--\n",
+    "output' option to specify a filename.\n",
+    "\n",
+    "Default population parameters for Anopheles gambiae:\n",
+    "Generation time: 0.09090909090909091\n",
+    "Population size: 6000000.0\n",
+    "Mutation rate: 5.49e-09\n",
+    "Recombination rate: 1.044e-08\n",
+    "\n",
+    "...\n",
+    "...\n",
+    "...\n",
+    "```"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "dated-louis",
+   "metadata": {},
+   "source": [
+    "### Initiating a Pull Request for your species\n",
+    "Okay if we are happy enough with our species we are ready to commit all our changes to our branch and start a pull request on github. First we double check we are on our new branch\n",
+    "and things are as they should be\n",
+    "```\n",
+    "git status\n",
+    "```\n",
+    "and if all good we are ready to commit with\n",
+    "```\n",
+    "git add stdpopsim/catalog/AnoGam/*.py\n",
+    "git commit -am \"added mosquito genome woot\"\n",
+    "\n",
+    "```\n",
+    "depending on how well you follow official style, the pre-commit hooks \n",
+    "that call the auto-formatting tools `black` and `flake8` \n",
+    "may or may not reformat your code. if they do simply rerun `git commit`\n",
+    "and you should now successfully be able to commit your code. \n",
+    "\n",
+    "Having done that the next step is to push your code to your fork\n",
+    "on github, from which you can start the PR. \n",
+    "```\n",
+    "git push --set-upstream origin mosquito\n",
+    "```\n",
+    "if successful this will return some output including an URL address to\n",
+    "a github page that will allow you to immediately open a new Pull Request,\n",
+    "asking the maintainers of `stdpopsim` to look over your code. e.g.,\n",
+    "\n",
+    "![Figure 3](gh3.png)\n",
+    "\n",
+    "in the little text box shown it's customary to write a brief description\n",
+    "of the work you've done, the features you have added, etc.. \n",
+    "Github offers a wide array of tools for referencing issues and tasks \n",
+    "through the Pull Request system, and you'll learn these with experience. \n",
+    "\n",
+    "\n",
+    "-------------------------\n",
+    "\n",
+    "### Start QC Issue\n",
+    "\n",
+    "Finally start a new issue using the Species QC template seen below\n",
+    "![Figure 2](gh2.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "abandoned-organ",
+   "metadata": {},
+   "source": [
+    "### A note about tests\n",
+    "`stdpopsim` in part guarantees code quality through the use of unit testing. Basic sanity tests for our new species will be completed through QC over in the `tests/test_AnoGam.py` file our `add-species` utility stubbed out for us.\n",
```

Comment:
I think we need to bring testing up earlier - we'll save developers a lot of hassle if they get used to running tests early, definitely before making a PR. The test stub is your friend when making a new species definition, it'll catch lots of common mistakes.

Reply:
so question about the test stub before I start editing-- as generated it looks like the test stub is for QC? is that not right?

Reply:
Yes and no. We add the stub in to (a) provide basic tests to make sure that the model the contributor adding makes sense (like checking they've actually filled out stuff, but not really thinking too hard about the values).

It then provides *placeholders* for the QC person later. These QC tests aren't actually run, because they are marked as "skip". So, the test stub gets commited to the repo in the first pass, and then someone else comes along and filles out the values in the tests and deletes the "skip" lines.

So, all you have to do as a QC person is find the right values **without** looking at the code in a catalog.

Reply:
do you reckon just showing how to run the tests is enough? is there a test you would like users to add? looks like the only tests set to run are
- `test_ensembl_id()`
- `test_name()`
- `test_common_name()`
- and `test_mutation_rate()`

the rest are set to skip until QC and all these pass out of the box

Reply:
No, no need to add tests, just run them. Try adding a species first without doing any updates and seeing how things fail. Or put in a Citation without a URL or something. 

The idea is this catches simple errors on the first pass and save some time in review.

Reply:
ah i see what you are saying. yes can do this for sure.